src/element_promotion/PromoteElementImpl.C

```diff
@@ -0,0 +1,702 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+
+#include <element_promotion/PromoteElementImpl.h>
+#include <element_promotion/PromotedPartHelper.h>
+#include <element_promotion/HexNElementDescription.h>
+
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_mesh/base/Field.hpp>
+#include <stk_mesh/base/FieldParallel.hpp>
+#include <stk_mesh/base/GetBuckets.hpp>
+#include <stk_mesh/base/GetEntities.hpp>
+#include <stk_mesh/base/Part.hpp>
+#include <stk_mesh/base/Selector.hpp>
+#include <stk_mesh/base/FEMHelpers.hpp>
+#include <stk_mesh/base/HashEntityAndEntityKey.hpp>
+#include <stk_mesh/base/CreateEdges.hpp>
+#include <stk_mesh/base/CreateFaces.hpp>
+#include <stk_search/CoarseSearch.hpp>
+#include <stk_search/IdentProc.hpp>
+#include <stk_io/StkMeshIoBroker.hpp>
+#include <stk_topology/topology.hpp>
+#include <stk_util/parallel/ParallelReduce.hpp>
+#include <stk_util/parallel/CommSparse.hpp>
+#include <stk_util/parallel/ParallelComm.hpp>
+#include <stk_util/util/ReportHandler.hpp>
+
+#include <boost/functional/hash/hash.hpp>
```

Comment:
Is this used?

Reply:
No, I got rid of the need for it with this PR but forgot to remove the header.  Ill fix that.